File: csharp/KaranlikMadde.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NDP_Odev4
{
    internal class KaranlikMadde : KuantumNesnesi, IKritik
    {
        public KaranlikMadde(string id, int stabilite, int tseviyesi):base(id, stabilite, tseviyesi)
        {
            NesneID = id;
            Stabilite = stabilite;
            TehlikeSeviyesi = tseviyesi;
        }

        public override void AnalizEt()
        {
            int yeniDeger = Stabilite - 15;

            if (yeniDeger <= 0)
            {
                throw new KuantumCokusuException(NesneID);
            }
            Stabilite = yeniDeger;
        }

        public void AcilDurumSogutmasi()
        {   
            int yeniDeger = Stabilite + 50;
            if (!(yeniDeger > 0 && yeniDeger< 100))
            {
                Console.WriteLine("Sınır değer aşılıyor...");
                Stabilite = 100;
            }
            Stabilite = yeniDeger;
            Console.WriteLine($"Bu nesne soğutuldu:{Stabilite}");

        }
    }
}

File: javascript/KaranlikMadde.js
const KuantumNesnesi = require('./KuantumNesnesi');
const KuantumCokusuException = require('./KuantumCokusuException');

class KaranlikMadde extends KuantumNesnesi {
    constructor(id, stabilite, tseviyesi) {
        super(id, stabilite, tseviyesi);
    }
    
    AnalizEt() {
        let yeniDeger = this.Stabilite - 15;
        if (yeniDeger <= 0) {
            throw new KuantumCokusuException(this.NesneID);
        }
        this.Stabilite = yeniDeger;
        console.log("Karanlık madde titriyor...");
    }

    AcilDurumSogutmasi() {
        let yeniDeger = this.Stabilite + 50;
        
        if (!(yeniDeger > 0 && yeniDeger < 100)) {
            this.Stabilite = 100;
            console.log("Sınır değer aşılıyor... " + this.Stabilite);
        } else {
            this.Stabilite = yeniDeger;
            console.log("Bu nesne soğutuldu: " + this.Stabilite);
        }
    }
}

module.exports = KaranlikMadde;